packages/react-scheduler/src/ReactScheduler.js

```diff
@@ -300,11 +288,24 @@ if (!ExecutionEnvironment.canUseDOM) {
       isAnimationFrameScheduled = true;
       requestAnimationFrame(animationTick);
     }
-    return newCallbackId;
+    return scheduledCallbackConfig;
   };
 
-  cancelScheduledWork = function(callbackId: number) {
-    delete registeredCallbackIds[callbackId];
+  cancelScheduledWork = function(callbackConfig: CallbackConfigType) {
+    const previousCallbackConfig = callbackConfig.previousCallbackConfig;
+    const nextCallbackConfig = callbackConfig.nextCallbackConfig;
+    if (previousCallbackConfig) {
+      previousCallbackConfig.nextCallbackConfig = nextCallbackConfig;
+    }
+    if (nextCallbackConfig) {
+      nextCallbackConfig.previousCallbackConfig = previousCallbackConfig;
+    }
```

Comment:
If there is no next or previous callback, then we need to set the head and tail to null, I think?